### PR TITLE
Pgpkgmigrations

### PR DESCRIPTION
**Pull request by bitner**
## Technical Context
- **Related Issues:** N/A
- **Breaking Changes:** Yes — migration filenames are renamed from `pgstac.X.Y.Z.sql` / `pgstac.X.Y.Z-A.B.C.sql` to `pgstac--X.Y.Z.sql` / `pgstac--X.Y.Z--A.B.C.sql`. Any tooling or scripts that reference migration files by the old dotted naming convention will need to be updated. The `pypgstac migrate` CLI and Python API continue to work without changes.

---

## Description

This PR introduces `pgstac-migrate`, a new lightweight standalone Python package that owns all of PgSTAC's runtime migration logic, and backs `stageversion` / `makemigration` with [pgpkg](https://github.com/pjmd/pgpkg) for reproducible SQL assembly and diff-based migration generation.

### Migration filename canonicalization

All migration files are renamed to use the double-dash separator:
pgstac.0.9.10.sql → pgstac--0.9.10.sql
pgstac.0.9.10-0.9.11.sql → pgstac--0.9.10--0.9.11.sql

This matches the [standard PostgreSQL extension file naming convention](https://www.postgresql.org/docs/current/extend-extensions.html) (`<extension>--<version>.sql`, `<extension>--<from>--<to>.sql`), removes ambiguity between version separators and the dots in version numbers, and improves interoperability with extension tooling like pgpkg.

### New `pgstac-migrate` package

`src/pgstac-migrate/` is a focused package that can be installed independently of `pypgstac`:

- **CLI**: `pgstac-migrate migrate [--to VERSION] [--dry-run]`, plus `info`, `versions`, `plan`, `build-artifact`
- **Python API**: `pgstac_migrate.api.migrate(target, conninfo, ...)` — usable directly without the full pypgstac stack
- **Baked artifact**: migration SQL is assembled into a single `migrations.tar.zst` bundle at publish time. This ships at **74 KB** versus the **8.1 MB** of raw SQL that `pypgstac` previously bundled via a symlink — roughly a **110× reduction** in migration payload. Those bytes move entirely out of `pypgstac` and into `pgstac-migrate`.

### `pypgstac migrate` is maintained as a compatibility wrapper — with planned deprecation

`pypgstac migrate` continues to work exactly as before; `Migrate.run_migration()` now delegates to `pgstac_migrate.api.migrate()` internally. No changes to calling code are required. This compatibility surface will be maintained for the near term but **is planned for deprecation in a future minor release**. Users should begin migrating to `pgstac-migrate` CLI or API directly.

### pgpkg-backed `stageversion` and `makemigration`

The in-container scripts now use `pgpkg` for SQL assembly and migration diffing, replacing the previous approach that used `migra` against two live databases. The optional `PGPKG_LOCAL_REPO_DIR` / `PGPKG_REPO_DIR` environment variable allows testing against a local pgpkg checkout; default behavior resolves pgpkg from PyPI with no local setup required.

**Reviewers: please focus on** `src/pgstac-migrate/` (new package structure and API surface), `src/pypgstac/src/pypgstac/migrate.py` (delegation wiring), and the cross-surface parity test at `src/pgstac-migrate/tests/test_parity.py` which asserts both tools produce identical ordered SQL for all migration paths.

---

## Checklist
- [x] **Linting:** Ruff lint and format pass; ty type checks pass; all pre-commit hooks pass
- [x] **Tests:** All existing tests pass. New tests added:
  - `src/pgstac-migrate/tests/` — CLI delegation, API, version source, and cross-surface plan parity (7 parametrized cases asserting pgstac-migrate catalog and pypgstac MigrationPath produce identical SQL file sequences)
  - `src/pypgstac/tests/test_migrate_wrapper.py` — delegation and compat layer coverage
  - `scripts/container-scripts/test` wired to run pgstac-migrate tests in CI
- [x] **Edge Cases:** Parity test covers fresh install, single-hop, and multi-hop upgrade paths; dry-run mode verified; delegation wiring tested with monkeypatching to confirm correct kwargs forwarded
- [x] **Documentation:** `CHANGELOG.md` and `docs/src/release-notes.md` updated; `CONTRIBUTING.md` updated to reflect two PyPI packages and new release workflow; `CLAUDE.md` and `AGENTS.md` updated with pgstac-migrate architecture and migration-engineer workflow
- [x] **Accountability:** I can explain every line submitted

---

## AI tool usage
- [x] AI (Copilot) supported development of this PR. See our [policy about AI tool use](https://stac-utils.github.io/ai-contribution-policy).

> Copilot assisted with test scaffolding (test_parity.py, delegation tests), changelog entries, and pre-commit fix diagnosis. All generated content was reviewed, run, and validated before inclusion.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -226,3 +226,29 @@ jobs:
...
+      url: https://pypi.org/p/pgstac-migrate
+    steps:
+      - name: Checkout
+        uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2
+      - name: Setup Python
+        uses: actions/setup-python@a309ff8b426b58ec0e2a45f0f869d46889d02405 # v6.2.0
+        with:
+          python-version: "3.x"
+      - name: Install build
+        working-directory: /home/runner/work/pgstac/pgstac/src/pgstac-migrate
```
Comment (gadomski) [inline .github/workflows/release.yml]: Can we just make these relative paths so they don't encode specifics of how Github Actions work?


```suggestion
        working-directory: src/pgstac-migrate
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -226,3 +226,29 @@ jobs:
...
+    environment:
+      name: pypi
+      url: https://pypi.org/p/pgstac-migrate
+    steps:
+      - name: Checkout
+        uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2
+      - name: Setup Python
+        uses: actions/setup-python@a309ff8b426b58ec0e2a45f0f869d46889d02405 # v6.2.0
+        with:
+          python-version: "3.x"
```
Comment (gadomski) [inline .github/workflows/release.yml]: This could/should be `setup-uv` per the copilot instructions

File: scripts/container-scripts/test [hunk 3]
```diff
@@ -66,31 +66,36 @@ EOSQL
...
 }
 
 function test_formatting(){
     cd $SRCDIR/pypgstac
 
     echo "Running ruff"
-    ruff check src/pypgstac tests
-    ruff format --check src/pypgstac tests
+    uv run --extra dev ruff check src/pypgstac tests
+    uv run --extra dev ruff format --check src/pypgstac tests
```
Comment (gadomski) [inline scripts/container-scripts/test]: Not for this PR, but we should prefer [dependency groups](https://docs.astral.sh/uv/concepts/projects/dependencies/#dependency-groups) over extras, as it keeps the packaging story cleaner.

File: scripts/makemigration [hunk 4]
```diff
@@ -12,7 +12,7 @@ Options:
   -f, --from VERSION      Source base version.
   -t, --to VERSION        Target base version.
   -o, --overwrite         Replace an existing migration file.
-  -d, --debug             Print the generated migra SQL before wrapping it.
+    -d, --debug             Print the generated wrapped migration SQL after creation.
```
Comment (gadomski) [inline scripts/makemigration]:
```suggestion
  -d, --debug             Print the generated wrapped migration SQL after creation.
```

File: src/pgstac-migrate/README.md
Comment (gadomski) [inline src/pgstac-migrate/README.md:1]: Since this is intended to be published to PyPI, the README should include instructions for using it from there?

File: src/pypgstac/src/pypgstac/migrate.py
Comment (gadomski) [inline src/pypgstac/src/pypgstac/migrate.py:1]: Are we intending on removing this functionality, eventually, in favor of the standalone package? If so we should throw some deprecation warnings on import (IMO).

File: src/pypgstac/pyproject.toml [hunk 5]
```diff
@@ -22,6 +22,7 @@ dependencies = [
     "fire>=0.7.0",
     "hydraters>=0.1.0",
     "orjson>=3.11.0",
+    "pgstac-migrate>=0.9.11.dev0,<0.10",
```
Comment (gadomski) [inline src/pypgstac/pyproject.toml]: Any reason to have the ceiling on the dependency?

**Review by gadomski — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (bitner) [reply .github/workflows/release.yml]: OK, everything should be using setup-uv now

File: scripts/container-scripts/test (hunk 3 above)
Comment (bitner) [reply scripts/container-scripts/test]: Went ahead and changed that here.

File: src/pypgstac/src/pypgstac/migrate.py
Comment (bitner) [reply src/pypgstac/src/pypgstac/migrate.py:1]: Added a deprecation warning.

File: src/pypgstac/pyproject.toml (hunk 5 above)
Comment (bitner) [reply src/pypgstac/pyproject.toml]: Nope. Ceiling removed.

**Review by gadomski — APPROVED** [review]
